File: luk_sdk/android/src/main/java/com/example/luk_sdk/LukSdkPlugin.java
package com.example.luk_sdk;
//package com.cftech.gamesdk;

import androidx.annotation.NonNull;

import io.flutter.Log;
import io.flutter.embedding.engine.plugins.FlutterPlugin;
import io.flutter.embedding.engine.plugins.activity.ActivityAware;
import io.flutter.embedding.engine.plugins.activity.ActivityPluginBinding;
import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;


import com.cftech.gamelibrary.CFGameSDK;
import com.cftech.gamelibrary.interfaces.ICFGameListCallback;
import com.cftech.gamelibrary.module.CFGameList;
//import com.cftech.gamelibrary.CFGameSDKInternal;
//import com.cftech.gamelibrary.engine.CFGameService;
//import com.cftech.gamelibrary.event.OpenApiEvent;
//import com.cftech.gamelibrary.interfaces.ICFGameListCallback;
//import com.cftech.gamelibrary.module.CFGameList;
//import com.cftech.gamelibrary.utils.CFDensityUtil;
//import com.cftech.gamelibrary.utils.CFFP;
//import com.cftech.gamelibrary.view.dialog.CFGameListDialog;
//import com.cftech.gamelibrary.view.webview.CFGWebView;

import java.util.List;
import java.util.Map;
import java.util.ArrayList;
import java.util.HashMap;

import android.app.Activity;
import android.content.Context;


/** LukSdkPlugin */
public class LukSdkPlugin implements FlutterPlugin, MethodCallHandler, ActivityAware {
  /// The MethodChannel that will the communication between Flutter and native Android
  ///
  /// This local reference serves to register the plugin with the Flutter Engine and unregister it
  /// when the Flutter Engine is detached from the Activity
  private MethodChannel channel;
  private Activity activity;
  private String TAG = "LukSdkPlugin";

  private String errMsg = "msg";
  private String errCode = "code";
  private String mtoken = "token";
  private String mgameList = "gameList";

  @Override
  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "luk_sdk");
    channel.setMethodCallHandler(this);

  }

  @Override
  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {

    if (call.method.equals("getPlatformVersion")) {
      result.success("Android " + android.os.Build.VERSION.RELEASE);
    } else if(call.method.equals("setupSDK")) {
        setupSDK(call,result);
    } else if(call.method.equals("setUserInfo")){
        setUserInfo(call,result);
    } else if(call.method.equals("getGameList")){
        getGameList(call,result);
    } else if(call.method.equals("releaseSDK")){
        releaseSDK(call,result);
    } else result.notImplemented();
  }


  @Override
  public void onDetachedFromEngine(@NonNull FlutterPluginBinding binding) {
    channel.setMethodCallHandler(null);
  }

  public void setupSDK(MethodCall call, Result result){
        int appid = Integer.parseInt(call.argument("appId"));
        String language =  call.argument("language");
        boolean isProduct = call.argument("isProduct");

    CFGameSDK.initSDK(this.activity.getApplication(),appid,  language, isProduct);

  }

  public void setUserInfo(MethodCall call, Result result){

    String userId = call.argument("userId");
    String userCode =  call.argument("userCode");

    CFGameSDK.setUserInfo(userId,userCode, new CFGameSDK.ICFLoginCallback() {

      @Override
      public void onLoginSuccess() {
        Log.i(TAG, "on Login Success.");
        Map result = new HashMap<>();
        result.put(errCode, Integer.toString(0));
        result.put(errMsg, "登录成功");
        if(channel != null){
          channel.invokeMethod("onLoginSuccessRes", result);
        }
      }

      @Override
      public void onLoginFail(int code, String msg) {
        Log.i(TAG, "on Login Fail.");

        Map result = new HashMap<>();
        result.put(errCode,code);
        result.put(errMsg, msg);

        if(channel != null){
          channel.invokeMethod("onLoginFailRes", result);
        }
      }

      @Override
      public void onRefreshToken(String token) {
        Log.i(TAG, "on Refresh Token." + token);

        Map result = new HashMap<>();
        result.put(errCode, 0);
        result.put(mtoken, token);
        result.put(errMsg, "success");

        if(channel != null){
          channel.invokeMethod("onRefrshToken", result);
        }
      }
    });
  }
  public void getGameList(MethodCall call, Result result){
    CFGameSDK.getGameList(new ICFGameListCallback() {

      @Override
      public void onSuccess(List<CFGameList.GameInfo> gameList) {
        Log.i(TAG, "get Game List success.");

        List<Map> newGameList = new ArrayList<Map>();
        if (!gameList.isEmpty()) {
          for (CFGameList.GameInfo gameInfo : gameList) {
              Map map =  new HashMap<>();
              map.put("g_id",gameInfo.g_id);
              map.put("g_name",gameInfo.g_name);
              map.put("g_icon",gameInfo.g_icon);
              map.put("g_url",gameInfo.g_url);
              newGameList.add(map);
          }
        }


        Map result = new HashMap<>();
        result.put(errCode, 0);
        result.put(mgameList, newGameList);

        Log.i(TAG, "get Game List success." + result);
        if(channel != null){
          channel.invokeMethod("onGetGameSuccess", result);
        }
      }

      @Override
      public void onError(int code, String errorMsg) {
        Map result = new HashMap<>();
        result.put(errCode, code);
        result.put(errMsg, errorMsg);

        if(channel != null){
          channel.invokeMethod("onGetGameFail", result);
        }
      }
    });
  }

  public void releaseSDK(MethodCall call, Result result){

    CFGameSDK.releaseSDK();
  }

  @Override
  public void onAttachedToActivity(@NonNull ActivityPluginBinding activityPluginBinding) {
    this.activity = activityPluginBinding.getActivity();

  }

  @Override
  public void onDetachedFromActivityForConfigChanges() {

  }

  @Override
  public void onReattachedToActivityForConfigChanges(@NonNull ActivityPluginBinding activityPluginBinding) {

  }

  @Override
  public void onDetachedFromActivity() {

  }
}

File: luk_sdk/ios/Classes/LukSdkPlugin.m
#import "LukSdkPlugin.h"
#import <CFGameSDK/CFGameSDK.h>
#import <CFGameSDK/CFGameModel.h>

@interface LukSdkPlugin() <CFGameSDKLoginDelegate>


@end



@implementation LukSdkPlugin {
    FlutterMethodChannel *_channel;
}

const NSString *_errCode = @"code";
const NSString *_token = @"token";
const NSString *_msg   = @"msg";
const NSString *_gameList = @"gameList";

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {

  FlutterMethodChannel* channel = [FlutterMethodChannel
      methodChannelWithName:@"luk_sdk"
            binaryMessenger:[registrar messenger]];
  LukSdkPlugin* instance = [[LukSdkPlugin alloc] initWithChannel:channel];
  [registrar addMethodCallDelegate:instance channel:channel];


}

- (instancetype)initWithChannel:(FlutterMethodChannel *)channel {
    self = [super init];
    if (self) {
        _channel = channel;
    }
    return self;
}



- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    result([@"iOS" stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
  }else if([@"setupSDK" isEqualToString:call.method]){
      [self setupSDK:call result:result];
  }else if([@"setUserInfo" isEqualToString:call.method]){
      [self setUserInfo:call result:result];
  }else if([@"getGameList" isEqualToString:call.method]) {
      [self getGameList];
  }else if([@"releaseSDK" isEqualToString:call.method]) {
      [self releaseSDK];
  }else{
    result(FlutterMethodNotImplemented);
  }
}


// 初始化方法
- (void)setupSDK:(FlutterMethodCall*)call result:(FlutterResult)result{

    NSString *appid = call.arguments[@"appId"];
    NSString *language = call.arguments[@"language"];
    BOOL isProduct = [call.arguments[@"isProduct"] boolValue];

    NSLog(@"userid %@,usercode %@ isProduct %d",appid,language,isProduct);

    [CFGameSDK setUpSDKWithApplication:UIApplication.sharedApplication appId:appid language:language isProduct:isProduct];

}



//  登录方法
- (void)setUserInfo:(FlutterMethodCall *)call result:(FlutterResult)result{

    NSString *userId = call.arguments[@"userId"];
    NSString *userCode = call.arguments[@"userCode"];

    NSLog(@"userid %@,usercode %@",userId,userCode);
    [CFGameSDK setUserInfo:userId code:userCode loginCallBack:self];
}


#pragma mark -- CFGameSDKLoginDelegate

//  用户登录失败回调
- (void)onLoginFailCode:(int)code message:(NSString *)msg {
    NSLog(@"登录失败");
    NSDictionary *result = @{
        _errCode:@(code),_msg:msg};

    if(_channel != nil){
        [_channel invokeMethod:@"onLoginFailRes" arguments:result];
    }
}

//  用户登录成功回调
- (void)onLoginSuccess {
    NSLog(@"登录成功");
    NSDictionary *result = @{
            _errCode:@(0),_msg:@"登录成功"
    };

    if(_channel != nil){
        [_channel invokeMethod:@"onLoginSuccessRes" arguments:result];
    }
}

//  token更新后回调，接入方通常无需关注此接口
- (void)onRefreshToken:(NSString *)token {
    NSLog(@"refresh token %@",token);
    NSDictionary *result = @{_errCode:@(0),_token:token};
    
    if(_channel != nil){
        [_channel invokeMethod:@"onRefrshToken" arguments:result];
    }
}

- (void)getGameList{
    NSLog(@"getGameList");
    [CFGameSDK getGameListWithSuccess:^(NSArray<CFGameModel *> * _Nonnull gameList) {
    
        NSLog(@"getGameList success");
        
        if (gameList && gameList.count > 0) {
            
            NSMutableArray *gameArr = [NSMutableArray array];
            for (CFGameModel *Model in gameList) {
                NSDictionary *dict = @{@"g_id":@(Model.g_id),
                                       @"g_name":Model.g_name,
                                       @"g_icon":Model.g_icon,
                                       @"g_url":Model.g_url};
                [gameArr addObject:dict];
            }
            NSArray *gameNewArr = [NSArray arrayWithArray:gameArr];

            NSDictionary *result = @{_errCode:@(0),_msg:@"success",_gameList:gameNewArr};

            if(self->_channel != nil){
                [self->_channel invokeMethod:@"onGetGameSuccess" arguments:result];
            }
        }else{
            if(self->_channel != nil){
                NSDictionary *result = @{_errCode:@(999),_msg:@"list is null"};
                [self->_channel invokeMethod:@"onGetGameFail" arguments:result];
            }
        }
        
        
    } failure:^(int code, NSString * _Nonnull msg) {
        NSLog(@"getGameList fail %@",msg);
        if(self->_channel != nil){
            NSDictionary *result = @{_errCode:@(code),_msg:msg};
            [self->_channel invokeMethod:@"onGetGameFail" arguments:result];
        }
    }];
    
}

- (void)releaseSDK{
    [CFGameSDK releaseSDK];
}










@end
